modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/SalesforceObjectEntryManagerImpl.java

```diff
@@ -482,7 +482,12 @@ private JSONObject _toJSONObject(
 				objectField.getExternalReferenceCode(),
 				Objects.equals(value, StringPool.BLANK) ? null : value);
 
-			if (Objects.equals(
+			String externalReferenceCode =
+				objectDefinition.getExternalReferenceCode();
+
+			if (Validator.isNotNull(externalReferenceCode) &&
+				externalReferenceCode.endsWith("__c") &&
```

Comment:
@guilhermedcamacho,

I believe we should define `"__c"` as a constant (`CUSTOM_OBJECT_SUFFIX`) in a class like SalesForceConstants or something like that.

I found that `__c` could be applied for custom field as well ([see](https://salesforce.stackexchange.com/questions/101844/what-are-the-object-and-field-name-suffixes-that-salesforce-uses-such-as-c-an)), but we can define one constant for now just for this use case.

What do you think?